digServiceExtractor/names_helper.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# @Author: ZwEin
+# @Date:   2016-10-05 16:01:52
+# @Last Modified by:   ZwEin
+# @Last Modified time: 2016-10-06 22:40:34
+
+import os
+import json
+import string
+import pygtrie as trie
+from digDictionaryExtractor.populate_trie import populate_trie
+from digDictionaryExtractor.name_dictionary_extractor import get_name_dictionary_extractor
+from digExtractor.extractor_processor import ExtractorProcessor
+from digCrfTokenizer.crf_tokenizer import CrfTokenizer
+
+def load_names(path=os.path.join(os.path.dirname(__file__), 'names.json')):
+    return json.load(open(path, 'r'))
+
+names = load_names()
+names_trie = populate_trie(iter(names))
+
+def tokenize(raw):
+    t = CrfTokenizer()
+    t.setRecognizeHtmlEntities(True)
+    t.setRecognizeHtmlTags(True)
+    t.setSkipHtmlTags(True)
+    t.setRecognizePunctuation(True)
+    tokens = t.tokenize(raw)
+    return tokens
+
+def extract(doc):
```

Comment:
Maybe we should discuss this.  The service extractor should accept tokens, not accept text and then create tokens.  We're trying to have separation of concerns here and not duplicate functionality.  If you're starting from text, you should have a TokenizerExtractor to generate the tokens and then use the output of the TokenizerExtractor as input to the ServiceExtractor.

https://github.com/usc-isi-i2/dig-memex-2016/blob/conda-workflow/conda-workflow/extractor_workflow_sans_spark.py#L143